src/Compilers/Core/Portable/FileSystem/PathUtilities.cs

```diff
@@ -53,6 +57,34 @@ public static string TrimTrailingSeparators(string s)
             return s;
         }
 
+        /// <summary>
+        /// Ensures a trailing directory separator character
+        /// </summary>
+        public static string AddTrailingSeparator(string s)
```

Comment:
I would rename this method "EnsureTrailingSeparator". The "Add" prefix implies it will always do work (aka allocate).